fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PruneFileScanPartition.java

```diff
@@ -55,22 +55,26 @@ public class PruneFileScanPartition extends OneRewriteRuleFactory {
 
     @Override
     public Rule build() {
-        return logicalFilter(logicalFileScan()).when(p -> p.child().getSelectedPartitions() == null).thenApply(ctx -> {
-            LogicalFilter<LogicalFileScan> filter = ctx.root;
-            LogicalFileScan scan = filter.child();
-            ExternalTable tbl = scan.getTable();
-            SelectedPartitions selectedPartitions = new SelectedPartitions(0, Maps.newHashMap(), false);
+        return logicalFilter(logicalFileScan()).when(p -> !p.child().getSelectedPartitions().isPartitionPruned())
+                .thenApply(ctx -> {
+                    LogicalFilter<LogicalFileScan> filter = ctx.root;
+                    LogicalFileScan scan = filter.child();
+                    ExternalTable tbl = scan.getTable();
+                    SelectedPartitions selectedPartitions = new SelectedPartitions(0, Maps.newHashMap(), false);
```

Comment:
all class used in Nereids plan stage, should be immutable. so u should use `ImmutableMap.of()` instead of `Maps.newHashMap()`. And use `ImmutableMap.copy()` in `SelectedPartitions`'s constructor. BTW u should check all parameters non-null by `Objects.requireNonNull(param, msg)` in `SelectedPartitions`'s constructor too.